api/plugin.go

```diff
@@ -0,0 +1,242 @@
+	}
+
+	// Plugin-specific validations
+	if len(req.Messages) != 1 {
+		return fmt.Errorf("plugin signing requires exactly one message hash")
+	}
+	if len(req.Transactions) != 1 {
+		return fmt.Errorf("plugin signing requires exactly one transaction")
+	}
+
+	// Get policy
+	policyPath := fmt.Sprintf("policies/%s.json", req.PolicyID)
+	policyContent, err := s.blockStorage.GetFile(policyPath)
+	if err != nil {
+		return fmt.Errorf("fail to read policy file, err: %w", err)
+	}
+
+	var policy types.PluginPolicy
+	if err := json.Unmarshal(policyContent, &policy); err != nil {
+		return fmt.Errorf("fail to unmarshal policy, err: %w", err)
+	}
+
+	// Validate policy matches plugin
+	if policy.PluginID != req.PluginID {
+		return fmt.Errorf("policy plugin ID mismatch")
+	}
+
+	// Parse payroll policy
+	var payrollPolicy types.PayrollPolicy
+	if err := json.Unmarshal(policy.Policy, &payrollPolicy); err != nil {
+		return fmt.Errorf("fail to unmarshal payroll policy, err: %w", err)
+	}
+
+	// Validate transaction matches policy
+	if err := validateTransaction(req.Transactions[0], payrollPolicy); err != nil {
+		return fmt.Errorf("transaction validation failed: %w", err)
+	}
+
+	// Validate message hash matches transaction
+	txHash, err := calculateTransactionHash(req.Transactions[0])
+	if err != nil {
+		return fmt.Errorf("fail to calculate transaction hash: %w", err)
+	}
+	if txHash != req.Messages[0] {
+		return fmt.Errorf("message hash does not match transaction hash. expected %s, got %s", txHash, req.Messages[0])
+	}
+
+	// Reuse existing signing logic
+	result, err := s.redis.Get(c.Request().Context(), req.SessionID)
+	if err == nil && result != "" {
+		return c.NoContent(http.StatusOK)
+	}
+
+	if err := s.redis.Set(c.Request().Context(), req.SessionID, req.SessionID, 30*time.Minute); err != nil {
+		s.logger.Errorf("fail to set session, err: %v", err)
+	}
+
+	filePathName := req.PublicKey + ".bak"
+	content, err := s.blockStorage.GetFile(filePathName)
+	if err != nil {
+		return fmt.Errorf("fail to read file, err: %w", err)
+	}
+
+	_, err = common.DecryptVaultFromBackup(req.VaultPassword, content)
+	if err != nil {
+		return fmt.Errorf("fail to decrypt vault from the backup, err: %w", err)
+	}
+
+	buf, err := json.Marshal(req)
+	if err != nil {
+		return fmt.Errorf("fail to marshal to json, err: %w", err)
+	}
+
+	ti, err := s.client.EnqueueContext(c.Request().Context(),
+		asynq.NewTask(tasks.TypeKeySign, buf),
+		asynq.MaxRetry(-1),
+		asynq.Timeout(2*time.Minute),
+		asynq.Retention(5*time.Minute),
+		asynq.Queue(tasks.QUEUE_NAME))
+
+	if err != nil {
+		return fmt.Errorf("fail to enqueue task, err: %w", err)
+	}
+
+	return c.JSON(http.StatusOK, ti.ID)
+}
+
+func (s *Server) CreatePluginPolicy(c echo.Context) error {
+	var policy types.PluginPolicy
+	if err := c.Bind(&policy); err != nil {
+		return fmt.Errorf("fail to parse request, err: %w", err)
+	}
+
+	policyPath := fmt.Sprintf("policies/%s.json", policy.ID)
```

Comment:
1. transaction policy need to signed with EDDSA key ,and server need to use it's public key to verify the signature

Reply:
Yes, signature verification for the policy doc is on the pending list in the PR description

Why eddsa? Most of the vault identification flows seem to use ECDSA.

Reply:
EDdSA is faste then ECDSA , ECDSA signing take 10 rounds , while EdDSA takes 4